### PR TITLE
Add note on active_record.encryption.support_sha1_for_non_deterministic_encryption

### PR DESCRIPTION
**Pull request by ezekg**
Follow up to #49587. Please let me know if anything is incorrect here. I'm still piecing together the situation discussed in https://github.com/rails/rails/issues/42922 and https://github.com/rails/rails/issues/48204, so if @boomer196 and @jorgemanrubia could review as well, that'd be great. I want to make sure the information here is as accurate as possible since this could cause a production outage if not handled correctly.

### CODE REVIEW

**Comment by rafaelfranca** [comment]
Thanks! I asked @jorgemanrubia's review

**Comment by boomer196** [comment]
It looks accurate to me. Thanks!

File: guides/source/upgrading_ruby_on_rails.md [hunk 1]
```diff
@@ -335,6 +335,22 @@ versions, there are two scenarios to consider:
     config.active_record.encryption.hash_digest_class = OpenSSL::Digest::SHA256
     ```
 
+See the [Configuring Rails Applications](https://guides.rubyonrails.org/configuring.html#config-active-record-encryption-hash-digest-class)
```
Comment (adrianna-chang-shopify) [inline guides/source/upgrading_ruby_on_rails.md:338]: Nit: We can shorten these links up.
```suggestion
See the [Configuring Rails Applications](configuring.html#config-active-record-encryption-hash-digest-class)
```

File: guides/source/upgrading_ruby_on_rails.md [hunk 2]
```diff
@@ -335,6 +335,22 @@ versions, there are two scenarios to consider:
...
+guide for more information on `config.active_record.encryption.hash_digest_class`.
+
+In addition, a new configuration [`config.active_record.encryption.support_sha1_for_non_deterministic_encryption`](https://guides.rubyonrails.org/configuring.html#config-active-record-encryption-support-sha1-for-non-deterministic-encryption)
```
Comment (adrianna-chang-shopify) [inline guides/source/upgrading_ruby_on_rails.md:341]:
```suggestion
In addition, a new configuration [`config.active_record.encryption.support_sha1_for_non_deterministic_encryption`](configuring.html#config-active-record-encryption-support-sha1-for-non-deterministic-encryption)
```

File: guides/source/upgrading_ruby_on_rails.md [hunk 3]
```diff
@@ -335,6 +335,22 @@ versions, there are two scenarios to consider:
...
+was introduced to resolve [a bug](https://github.com/rails/rails/issues/42922) that caused some attributes to be
+encrypted using SHA-1 even when SHA-256 was configured via the aforementioned `hash_digest_class` configuration.
+
+By default, `config.active_record.encryption.support_sha1_for_non_deterministic_encryption` is disabled in
+Rails 7.1. If you have data encrypted in a version of Rails < 7.1 that you believe may be effected
```
Comment (adrianna-chang-shopify) [inline guides/source/upgrading_ruby_on_rails.md:346]:
```suggestion
Rails 7.1. If you have data encrypted in a version of Rails < 7.1 that you believe may be affected
```

**Review by adrianna-chang-shopify — COMMENTED** [review]
Thanks for adding to the docs! ❤️

File: guides/source/upgrading_ruby_on_rails.md (hunk 3 above)
Comment (ezekg) [reply guides/source/upgrading_ruby_on_rails.md:346]: This one always gets me.

File: guides/source/upgrading_ruby_on_rails.md (hunk 2 above)
Comment (ezekg) [reply guides/source/upgrading_ruby_on_rails.md]: Done.

**Review by jorgemanrubia — APPROVED** [review]
This is great. Thanks for adding it.

**Review by zzak — APPROVED** [review]

**Pull request merged**